src/Agent.Plugins/BuildDrop/BuildDropDropPlugin.cs

```diff
@@ -0,0 +1,234 @@
+            Guid projectId, 
+            int buildId, 
+            string artifactName,
+            PreprocessResult preprocResult,
+            CancellationToken token)
+        { 
+            // Upload to VSTS BlobStore, and associate the artifact with the build.
+            context.Output($"Upload artifact: {preprocResult.FullPath} to server for build: {buildId} at backend.");
+            BuildDropClient server = new BuildDropClient();
+            await server.UploadDropArtifactAsync(context, context.VssConnection, projectId, buildId, artifactName, preprocResult.FullPath, token);
+            context.Output($"Upload artifact finished.");
+        }
+
+        private Boolean IsUncSharePath(AgentTaskPluginExecutionContext context, string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return false;
+            }
+
+            Uri uri;
+            // Add try catch to avoid unexpected throw from Uri.Property.
+            try
+            {
+                if (Uri.TryCreate(path, UriKind.RelativeOrAbsolute, out uri))
+                {
+                    if (uri.IsAbsoluteUri && uri.IsUnc)
+                    {
+                        return true;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                context.Debug($"Can't determine path: {path} is UNC or not.");
+                context.Debug(ex.ToString());
+                return false;
+            }
+
+            return false;
+        }
+    }
+
+    // To be invoked by DownloadBuildArtifacts task
+    public class DownloadBuildDropTask : BuildDropTaskPluginBase
+    {
+        // Same as https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/DownloadBuildArtifacts/task.json
+        public override Guid Id => new Guid("61F2A582-95AE-4948-B34D-A1B3C4F6A737");
+
+        public override string Version => "0.137.0";
+
+        public override string Stage => "main";
+
+        protected override PreprocessResult CheckAndTransformTargetPath(
+            AgentTaskPluginExecutionContext context, string localPath, string artifactName)
+        { 
+
+            string fullPath = Path.GetFullPath(localPath);
+
+            bool isDir = Directory.Exists(fullPath);
+            if (!isDir)
+            {
+                Directory.CreateDirectory(fullPath);
+            }
+
+            PreprocessResult result = new PreprocessResult {
+                FullPath = fullPath
+            };
+
+            if (context.Inputs.TryGetValue(ArtifactEventProperties.BuildId, out string buildId) && 
+                UInt32.TryParse(buildId, out uint value) &&
+                value != 0)
+            {
+                context.Output($"Download from the specified build: #{ value }");
+                result.BuildId = value;
+            }
+            else
+            {
+                context.Output("Download from the current build.");
```

Comment:
i would like to print and set the current build id.

Reply:
done